phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -3529,6 +3566,19 @@ private MetaDataMutationResult mutateColumn(
         }
     }
 
+    private void invalidateServerMetadataCache(List<InvalidateServerMetadataCacheRequest> requests)
+            throws Throwable {
+        Properties properties = new Properties();
```

Comment:
Shouldn't we check the invalidate metadata cache config earlier and short circuit earlier before doing all the work ?

Reply:
Yes I believe we could!
But there are multiple places we are invalidating server cache from and there could be more in the future. Assuming we don't do a lot of DDL changes which invalidate the cache, maybe it is okay to keep the check in one place? We could move it to this method in MDEI instead of keeping it in CQSI.

Reply:
Yes, that's what I was getting to. Doing the check early. You could keep the check in CQSI also.

Reply:
> before doing all the work

@tkhurana By this, do you mean creating the list of invalidation requests?

Reply:
Okay I guess we can keep a boolean flag in MDEI itself with the config value which is set when the coproc is loaded.

Reply:
@palashc What I meant was we can check the config and return early before creating the server connection

Reply:
@tkhurana I see. Do you think keeping the boolean flag in MDEI and checking that before we create the invalidation requests would be even better?

Reply:
Yes, that should be even better.

Reply:
Done. Stuck to your previous suggestion to just keep it in one place before we create server connection.